pkg/test/framework/components/echo/calloptions.go

```diff
@@ -39,71 +88,40 @@ type CallOptions struct {
 	// port will be used (if feasible).
 	Scheme scheme.Instance
 
-	// If true, h2c will be used in HTTP requests
-	HTTP2 bool
-
-	// If true, HTTP/3 request over QUIC will be used.
-	// It is mandatory to specify TLS settings
-	HTTP3 bool
-
 	// Address specifies the host name or IP address to be used on the request. If not provided,
 	// an appropriate default is chosen for the target Instance.
 	Address string
 
-	// Path specifies the URL path for the HTTP(s) request.
-	Path string
-
 	// Count indicates the number of exchanges that should be made with the service endpoint.
 	// If Count <= 0, defaults to 1.
 	Count int
 
-	// Headers indicates headers that should be sent in the request. Ignored for WebSocket calls.
-	// If no Host header is provided, a default will be chosen for the target service endpoint.
-	Headers http.Header
-
 	// Timeout used for each individual request. Must be > 0, otherwise 5 seconds is used.
 	Timeout time.Duration
 
-	// Message to be sent if this is a GRPC request
-	Message string
-
-	// ExpectedResponse asserts this is in the response for TCP requests.
-	ExpectedResponse *wrappers.StringValue
-
-	// Method to send. Defaults to HTTP. Only relevant for HTTP.
-	Method string
-
-	// Use the custom certificate to make the call. This is mostly used to make mTLS request directly
-	// (without proxy) from naked client to test certificates issued by custom CA instead of the Istio self-signed CA.
-	Cert, Key, CaCert string
+	// HTTP settings.
+	HTTP HTTP
 
-	// Use the custom certificates file to make the call.
-	CertFile, KeyFile, CaCertFile string
+	// TCP settings.
+	TCP TCP
 
-	// Skip verify peer's certificate.
-	InsecureSkipVerify bool
+	// TLS settings.
+	TLS TLS
 
-	// FollowRedirects will instruct the call to follow 301 redirects. Otherwise, the original 301 response
-	// is returned directly.
-	FollowRedirects bool
+	// Message to be sent.
+	Message string
```

Comment:
I think this is part of TCP

Reply:
It seems to be used by multiple protocols